ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml

```diff
@@ -13,67 +13,85 @@
   when: xlc13.stat.islnk is defined
   tags: xlc13
 
-- name: Transfer and Extract XLC13
-  unarchive:
-    src: /Vendor_Files/aix/XLC/IBM_XL_C_C___FOR_AIX_V13.1.3_EMG.tar.gz
-    dest: /tmp
-    remote_src: no
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+- name: Check for Vendor Files
+  block:
+    - name: Check for Vendor Files
+      stat:
+        path: /Vendor_Files/aix/XLC/IBM_XL_C_C___FOR_AIX_V13.1.3_EMG.tar.gz
```

Comment:
This task checks for /Vendor_Files/aix/XLC/IBM_XL_C_C___FOR_AIX_V13.1.3_EMG.tar.gz on the target machine, where as it should check on the local machine. This can be done by adding `delegate_to: localhost`, but when doing that I get the following error

`fatal: [p9-aix1-ojdk06.osuosl.org]: FAILED! => {"changed": false, "module_stderr": "sudo: a password is required\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}`

Which makes me think that this particular task is not needed; I would prefer that the folllowing task throws an error if that file isnt present on the local machine, instead of skipped tasks, as a way of notifying the user

Reply:
What I have not done yet - but am planning to do is use a set_fact: statement to say - not ready at the end.

I feel it is important to do as much as possible - where/when not being there does not impact the rest of the playbook.

Nowhere is there documentation on what this tar file is, where it comes from. imho - it is a hack for a previous/prior location. So, I do not want to delete it because it is provider/location bound - but I don't want the playbook stopping in it's tracks.

IMHO: better to say (not ready, need xlc13) (so the not_ready fact needs to become a list of messages that get printed at the end.

As to locations: I want, once I am farther along, to have a template file of variables that can be set for where sources are coming from (e.g, vendor files, nimclient, etc..). But these are issues that come after the core logic runs end to end - and reports on system readiness. Idempotent!

Reply:
further, I read the task to mean the vendor files are on the target, not on the localhost (i.e., ansible control node).

Since the compiler is not installed - check if vendor files are there - opens block.
within the block - there are additional checks
when the tar file exists, unarchive it - register that we have install(p) images.

since we have images (in xlc16 i discovered the condition skipped which I presonally perfer - so the command: installp ... will become when _images is not skipped and rc==0

Also, a register value with a . (dot) in it - as the original has - is confusing imho - makes it look like a complex variable. So I have a later change with a new variable name as well.

In short, there are multiple conditions that need to be met within the block - not just the initial condition. That block also applies that condition is just background behavior irrelevant to the tasks within the block.